tests/unit/IonTextWriterTest.js

```diff
@@ -504,6 +504,37 @@
   )
 }`);
 
+      //depth tests
+      var depthTest = function(name, instructions, expectedDepth) {
+          suite[name] = function() {
+              var writeable = new ionTest.Writeable();
+              var writer = new ionTextWriter.TextWriter(writeable);
+              instructions(writer);
+              assert.equal(writer.depth(), expectedDepth);
+          }
+      }
+      depthTest('Stepping into a list results in a depth of 1.',
+          (writer) => {writer.stepIn(ion.IonTypes.LIST);}, 1);
+      depthTest('Stepping into a struct results in a depth of 1.',
+          (writer) => {writer.stepIn(ion.IonTypes.STRUCT);}, 1);
+      depthTest('Stepping into a sexp results in a depth of 1.',
+          (writer) => {writer.stepIn(ion.IonTypes.SEXP);}, 1);
+      depthTest('Stepping into 2 lists results in a depth of 2.',
+          (writer) => {writer.stepIn(ion.IonTypes.LIST); writer.stepIn(ion.IonTypes.LIST);}, 2);
```

Comment:
Is this copy-paste from the above unit tests, for the binary writer?  Can we do something to not copy-paste this by hand?

Reply:
Yea seems like generic writer tests as a new file could reduce alot of repitition that already exists.